sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/internal/ServiceInstanceIdResourceProvider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure.internal;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ConditionalResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.UUID;
+
+/**
+ * does not implement {@link ResourceProvider}, because it depends on all attributes discovered by
+ * the other providers.
+ */
+public final class ServiceInstanceIdResourceProvider implements ConditionalResourceProvider {
```

Comment:
Let's go ahead and move this to `opentelemetry-sdk-extension-incubator`. The incubator is currently bundled with the otel java agent, so the effect should be the same. But moving it to the incubator is the more conservative / accurate home for now given the compatibility guarantees for the stable autoconfigure artifact.

Reply:
yes, will do :smile:

Reply:
@jack-berg done :smile:

Reply:
I moved this to `io.opentelemetry.sdk.extension.incubator.resources.*`, but looks good. Thanks!